### PR TITLE
release: prepare v0.2.0-beta.2

### PR DESCRIPTION
**Pull request by hardcoreerik**
The beta documentation still points users to v0.2.0-beta.1 even though `main` now contains the radio ownership/scan refactor and its regression suite. Hardware testing of the release candidate also exposed two Tab5 regressions: continuous DSP plus visualizer analysis could starve Core 1's watched idle task, and RTL-SDR hot-plug could repaint the former startup overlay over an active FM dashboard.

This PR prepares the source record for v0.2.0-beta.2 and includes the narrow fixes exercised by the current flashed test build.

Changes:
- update README, M5Burner, and build-policy references to v0.2.0-beta.2
- add beta.2 release notes covering radio ownership, the shared scan engine, regression tests, and acceptance boundaries
- run the best-effort `rf_analysis` worker at idle priority so the watched Core 1 idle task gets scheduler time during continuous DSP/FFT load
- remove the obsolete revision-driven RTL-SDR status painter that could draw the old status strip and **Open SDR radio** button over an active dashboard after USB hot-plug
- allow unchanged C6 image/update inputs to reuse prior hardware evidence while still requiring exact-package source revision and SHA-256 verification

Root-cause evidence:
- retained task-watchdog core dumps showed `rtl_dsp` in FM phase processing while `rf_analysis` was concurrently processing FFT work on Core 1
- stack and heap evidence were healthy; the failure was CPU starvation rather than a demonstrated leak, corrupt allocation, or brownout
- the hot-plug overlay came from a legacy status-revision path that painted home-screen elements without current dashboard ownership

Validation:
- native ESP-IDF 5.5.4 Tab5 build completed successfully
- the built images flashed to the Tab5 with verified hashes
- `RTL_VIS OPEN phosphor` returned `active=1 view=phosphor source=1 frozen=0`
- the attached no-reset serial monitor recorded no watchdog, panic, or reboot during the observed Phosphor intervals
- a later health snapshot reported `uptime_ms=8001802`, `free_heap=26907720`, `min_free_heap=22101928`, `dma_free=87671`, `dma_min=87295`, `dma_largest=47104`, and `main_stack_hwm=2816`
- the two runtime files in this PR match the flashed bugfix worktree byte-for-byte
- existing optimized and sanitizer host tests and the previously recorded 10-cycle, 3-cycle, and 1-cycle Tab5 radio-scan runs remain part of the beta.2 validation record

Honest acceptance boundaries:
- the visualizer had been changed to Constellation by the final health snapshot, so the full elapsed interval is not claimed as a continuous Phosphor soak
- the exact unplugged-boot -> FM -> RTL-SDR hot-plug sequence has not yet been rerun on this flashed fix
- the project owner previously verified the unchanged C6 route by returning the reachable C6 to 2.14.6 and using OrcSDR to update it to 3.0.6; this PR does not alter that code or image
- this PR does not claim an exact-tag beta.2 bundle or M5Burner installation; those gates run after merge and tagging

After merge, the intended sequence is to create the immutable beta.2 tag on the merge commit, rebuild the exact-tag P4/C6 bundle, verify provenance and ZIP contents, privately install through M5Burner without erase, run final device checks, and only then publish the GitHub release and public M5Burner listing.

## Summary by CodeRabbit

- **Documentation**
  - Updated installation and release guidance to reference the current published beta through GitHub Releases and M5Burner.
  - Generalized hardware acceptance requirements across public beta releases, with conditional downgrade testing.
  - Added release notes for v0.2.0-beta.2, including scanning improvements, validation results, and release gates.

- **Bug Fixes**
  - Improved responsiveness during continuous radio DSP by running best-effort RF analysis at idle priority.

- **UI Changes**
  - Removed the RTL-SDR status banner and its “Open SDR radio” action from the home screen.

### CODE REVIEW

**Comment by hardcoreerik** [comment]
@coderabbitai review

File: docs/M5BURNER_RELEASE.md [hunk 1]
```diff
@@ -31,8 +31,8 @@ local test zip. It never flashes hardware or uploads a listing.
 
 ## Publication record and future-release gate
 
-`v0.2.0-beta.1` is publicly available through M5Burner and as a [GitHub
-prerelease](https://github.com/hardcoreerik/OrcSDR/releases/tag/v0.2.0-beta.1).
+`v0.2.0-beta.2` is publicly available through M5Burner and as a [GitHub beta
+release](https://github.com/hardcoreerik/OrcSDR/releases/tag/v0.2.0-beta.2).
```
[the remark this answers is not in the document]
Comment (hardcoreerik) [reply docs/M5BURNER_RELEASE.md]: Fixed in f019002. README now links to the release index and tells builders to substitute the tag from the published release. M5BURNER_RELEASE.md now describes published versions generically. The only beta.2-specific reference left in this document set is the release-note file itself, which explicitly says exact-tag publication gates remain pending.

**Comment by hardcoreerik** [comment]
@coderabbitai review

**Comment by hardcoreerik** [comment]
@coderabbitai review
